tests/components/nx584/test_binary_sensor.py

```diff
@@ -1,51 +1,62 @@
 """The tests for the nx584 sensor platform."""
-import unittest
 from unittest import mock
 
 from nx584 import client as nx584_client
 import pytest
 import requests
 
 from homeassistant.components.nx584 import binary_sensor as nx584
-from homeassistant.setup import setup_component
-
-from tests.common import get_test_home_assistant
+from homeassistant.setup import async_setup_component
 
 
 class StopMe(Exception):
     """Stop helper."""
 
-    pass
 
+@pytest.fixture
+def fake_zones():
+    """Fixture for fake zones.
 
-class TestNX584SensorSetup(unittest.TestCase):
-    """Test the NX584 sensor platform."""
+    Returns:
+        list: List of fake zones
+    """
+    return [
+        {"name": "front", "number": 1},
+        {"name": "back", "number": 2},
+        {"name": "inside", "number": 3},
+    ]
 
-    def setUp(self):
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        self._mock_client = mock.patch.object(nx584_client, "Client")
-        self._mock_client.start()
 
-        self.fake_zones = [
-            {"name": "front", "number": 1},
-            {"name": "back", "number": 2},
-            {"name": "inside", "number": 3},
-        ]
+@pytest.fixture
+def client(fake_zones):
+    """Fixture for client.
+
+    Args:
+        fake_zones (list): Fixture of fake zones
+
+    Yields:
+        MagicMock: Client Mock
+    """
+    _mock_client = mock.patch.object(nx584_client, "Client")
+    _mock_client.start()
 
-        client = nx584_client.Client.return_value
-        client.list_zones.return_value = self.fake_zones
-        client.get_version.return_value = "1.1"
-        self.addCleanup(self.tear_down_cleanup)
+    client = nx584_client.Client.return_value
+    client.list_zones.return_value = fake_zones
+    client.get_version.return_value = "1.1"
 
-    def tear_down_cleanup(self):
-        """Stop everything that was started."""
-        self.hass.stop()
-        self._mock_client.stop()
+    yield _mock_client
+
+    _mock_client.stop()
+
+
+@pytest.mark.usefixtures("client")
+class TestNX584SensorSetup:
+    """Test the NX584 sensor platform."""
 
+    @staticmethod
```

Comment:
Instead of a class with static methods, migrate it to just be functions outside of a class.

Reply:
I guess being in a class works too.

Reply:
For the future we should be consistent and not use classes in the tests.

Reply:
@MartinHjelmare added #42628 to remove the classes in the tests